### PR TITLE
Sprint 4 (dev->stg)

### PR DESCRIPTION
**Pull request by afrasier**
Promotes changes from the following PRs

* Domestic Filters #55 
* Internal Sharing #59 
* Saved Searches #62 
* Location models #63

### CODE REVIEW

File: talentmap_api/position/filters.py [hunk 1]
```diff
@@ -40,6 +40,8 @@ class PositionFilter(filters.FilterSet):
     bureau = filters.RelatedFilter(OrganizationFilter, name='bureau', queryset=Organization.objects.all())
     post = filters.RelatedFilter(PostFilter, name='post', queryset=Post.objects.all())
 
+    domestic = filters.BooleanFilter(name="is_overseas", lookup_expr="exact", exclude=True)
```
Comment (jseppi) [inline talentmap_api/position/filters.py]: For consistency, `is_domestic` might be a slightly better name for this field

Comment (jseppi) [inline talentmap_api/position/filters.py]: Same comment for `available` -> `is_available`. Not a big deal though.

File: .gitignore [hunk 2]
```diff
@@ -101,3 +101,6 @@ ENV/
 .mypy_cache/
 
 missing_coverage.sh
+
+# MacOS
+.DStore
```
Comment (burgwyn) [inline .gitignore]: This was me.  I'll get a PR to fix.

File: talentmap_api/position/filters.py (hunk 1 above)
Comment (afrasier) [inline talentmap_api/position/filters.py]: I agree, can you add an issue for that? I don't want to change these without some coordination on the front-end.

File: talentmap_api/common/xml_helpers.py [hunk 3]
```diff
@@ -123,13 +123,13 @@ def process_function(instance, item):
     return process_function
 
 
-def parse_boolean(field):
+def parse_boolean(field, true_values=["1", "True", "true", "Y", "T"]):
```
Comment (afrasier) [inline talentmap_api/common/xml_helpers.py]: Good to know, I didn't realize they were mutable (I thought it returned a copy each time). I'll modify this to set the defaults in the code block and have the argument override it.

File: talentmap_api/common/common_helpers.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+    # Instantiate the view
+    view = view.cls()
+
+    return view
+
+
+def validate_filters_exist(filter_list, filter_class):
+    for filter in filter_list.keys():
+        # If our filter is in the declared_filter list, it is a valid filter (these have some manual parsing)
```
Comment (afrasier) [inline talentmap_api/common/common_helpers.py]: I'll explain, keep your eyes peeled for a PR to dev.

File: talentmap_api/organization/views.py [hunk 5]
```diff
@@ -23,6 +23,24 @@ def get_queryset(self):
         return queryset
 
 
+class LocationView(ReadOnlyModelViewSet):
+    """
+    retrieve:
+    Return the given post.
+
+    list:
+    Return a list of all posts.
```
Comment (afrasier) [inline talentmap_api/organization/views.py]: 👍

File: talentmap_api/organization/views.py [hunk 6]
```diff
@@ -23,6 +23,24 @@ def get_queryset(self):
         return queryset
 
 
+class LocationView(ReadOnlyModelViewSet):
+    """
+    retrieve:
+    Return the given post.
```
Comment (afrasier) [inline talentmap_api/organization/views.py]: :+1:

File: talentmap_api/position/filters.py (hunk 1 above)
Comment (jseppi) [inline talentmap_api/position/filters.py]: https://github.com/18F/State-TalentMAP-API/issues/67

File: talentmap_api/common/xml_helpers.py (hunk 3 above)
Comment (jseppi) [inline talentmap_api/common/xml_helpers.py]: Yeah, it caught me off guard the first time I encountered it as well!

**Comment by afrasier** [comment]
@jseppi Issues should be resolved if you want to take another look!

**Pull request merged**